rpc/core/tx.go

```diff
@@ -133,3 +133,30 @@ func TxSearch(
 
 	return &ctypes.ResultTxSearch{Txs: apiResults, TotalCount: totalCount}, nil
 }
+
+// TxSearchMatchEvents allows you to query for multiple transactions results and match the
+// query attributes to a common event. It returns a
+// list of transactions (maximum ?per_page entries) and the total count.
+// More: https://docs.tendermint.com/v0.34/rpc/#/Info/tx_search
+func TxSearchMatchEvents(
+	ctx *rpctypes.Context,
+	query string,
+	prove bool,
+	pagePtr, perPagePtr *int,
+	orderBy string,
+	matchEvents bool,
+) (*ctypes.ResultTxSearch, error) {
+
+	// if index is disabled, return error
+	if _, ok := env.TxIndexer.(*null.TxIndex); ok {
+		return nil, errors.New("transaction indexing is disabled")
+	} else if len(query) > maxQueryLength {
+		return nil, errors.New("maximum query length exceeded")
+	}
+
```

Comment:
Similar to my other comment: isn't this check already handled in the `TxSearch` call?

Reply:
It is. I figured, we can return from here immediately if these conditions are met to reduce a function call.

Reply:
The function call is negligible in terms of practical impact. The practical impact of keeping these additional checks here is that they need to be kept in sync with the other checks (future developer effort).

I'd highly recommend removing them from both the `TxSearchMatchEvents` and `BlockSearchMatchEvents` methods and rather let the inner calls do the work.